lib/rust/parser/debug/LoadParser/LoadParser.java

```diff
@@ -0,0 +1,238 @@
+package org.enso.checkparser;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import org.enso.compiler.EnsoCompiler;
+import org.enso.compiler.codegen.AstToIr;
+import org.enso.compiler.core.IR;
+import org.enso.compiler.core.IR$Comment$Documentation;
+import org.enso.compiler.core.IR$Module$Scope$Definition;
+import org.enso.compiler.core.IR$Type$Ascription;
+import org.enso.syntax.text.AST;
+import org.enso.syntax.text.Shape;
+import org.enso.syntax2.Parser;
+import org.enso.syntax2.Tree;
+import org.graalvm.polyglot.Source;
+import scala.Function1;
+import scala.collection.immutable.List;
+
+class LoadParser implements FileVisitor<Path>, AutoCloseable {
+    private final File root;
+    private final Parser parser;
+    private final EnsoCompiler compiler;
+    private final Set<Path> visited = new LinkedHashSet<>();
+    private final Map<Path,Exception> failed = new LinkedHashMap<>();
+    private final Set<Path> irTested = new LinkedHashSet<>();
+    private final Map<Path,Exception> irFailed = new LinkedHashMap<>();
+    private final Set<Path> irDiff = new LinkedHashSet<>();
+
+    private LoadParser(File root) {
+        this.parser = Parser.create();
+        this.compiler = new EnsoCompiler();
+        this.root = root;
+    }
+
+    @Override
+    public void close() throws Exception {
+        parser.close();
+    }
+
+    public static void main(String[] args) throws Exception {
+        var root = new File(".").getAbsoluteFile();
+        try (LoadParser checker = new LoadParser(root)) {
+            checker.scan("distribution");
+            checker.scan("test");
```

Comment:
Yup, `test` is the right directory: https://github.com/enso-org/enso/pull/3611/commits/466ab24a8cb77172708b163add02670e7399ef13